### PR TITLE
[main][bugfix] Fix fullgraph padding bug in mtp eagle refactor

### PR DESCRIPTION
**Pull request by lilinsiman**
### What this PR does / why we need it?
The condition for determining padding in the fullgraph overlay with MTP and PCP has been modified to accommodate corner cases where the shape capture size is manually specified.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ut and tests

- vLLM version: v0.13.0
- vLLM main: https://github.com/vllm-project/vllm/commit/2f4e6548efec402b913ffddc8726230d9311948d

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -965,7 +965,7 @@ def _prepare_inputs(
             # the builders won’t need to add any extra padding.
             max_decode_tokens = self.scheduler_config.max_num_seqs * self.uniform_decode_query_len
             if self.compilation_config.cudagraph_mode.decode_mode() == CUDAGraphMode.FULL and \
-                uniform_decode and self.uniform_decode_query_len <= num_input_tokens <= max_decode_tokens:
+                uniform_decode and self.uniform_decode_query_len <= num_input_tokens <= min(max_decode_tokens, self.cudagraph_batch_sizes[-1]):
```
Comment (yiz-liu) [inline vllm_ascend/worker/model_runner_v1.py]: I suggest
```python
max_decode_tokens = min(self.scheduler_config.max_num_seqs * self.uniform_decode_query_len, self.cudagraph_batch_sizes[-1])
```
It's more resonable.

Comment (lilinsiman) [reply vllm_ascend/worker/model_runner_v1.py]: done

**Review by yiz-liu — APPROVED** [review]
